app/controllers/concerns/listings_toolkit.rb

```diff
@@ -4,14 +4,19 @@ module ListingsToolkit
 
   MANDATORY_FIELDS_FOR_UPDATE = %i[body_markdown title tag_list].freeze
 
-  def set_listing
+  included do
+    before_action :set_and_authorize_listing, only: %i[update]
+  end
+
+  def set_and_authorize_listing
```

Comment:
Updated this to include authorization. Every time `set_listing` was used it was implicitly in a `before_action` the first line of the method was `authorize @listing`.